### PR TITLE
Fix/44 `ActorSystem` lifecycle

### LINKED ISSUE #44: Support specifying an ActorSystem when [FixtureLifeCycle(LifeCycle.InstancePerTestCase)] is specified
nunit 3.13 added `[FixtureLifeCycle(LifeCycle.InstancePerTestCase)]` which should obviate the need for _code bombing_ when an `ActorSystem` is provided to the `TestKit` constructor.

**Describe the bug**
* Unable to specify `ActorSystem` in `TestKit` constructor

**To Reproduce**
* Try instantiating `TestKit` with an `ActorSystem`

See: https://github.com/akkadotnet/akka.net/pull/1092

### PR DESCRIPTION
**Pull request by Aaronontheweb**
Fixes #44

## Changes

Removed all of the old hacks we needed to do in order to make the NUnit testkit behave like the xUnit test kit (to provide isolated `ActorSystem`s on each test.)

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [x] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [x] This change follows the [Akka.NET Wire Compatibility Guidelines](https://getakka.net/community/contributing/wire-compatibility.html).
* [x] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).

cc @mchandschuh

### CODE REVIEW

File: src/Akka.TestKit.NUnit.Tests/Bugfix44Spec.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+﻿using Akka.Actor;
+using NUnit.Framework;
+
+namespace Akka.TestKit.NUnit.Tests;
+
+[TestFixture]
+public class Bugfix44Spec : TestKit
```
Comment (Aaronontheweb) [inline src/Akka.TestKit.NUnit.Tests/Bugfix44Spec.cs:7]: Repro spec for #44

File: src/Akka.TestKit.NUnit/TestKit.cs [hunk 2]
```diff
@@ -23,7 +23,6 @@ public class TestKit : TestKitBase, IDisposable
         private static readonly NUnitAssertions _assertions = new NUnitAssertions();
         private readonly Config _config;
         private readonly string _actorSystemName;
-        private bool _isFirstRun = true;
```
Comment (Aaronontheweb) [inline src/Akka.TestKit.NUnit/TestKit.cs:26]: No longer needed

File: src/Akka.TestKit.NUnit/TestKit.cs [hunk 3]
```diff
@@ -35,8 +34,6 @@ public class TestKit : TestKitBase, IDisposable
         public TestKit(ActorSystem system = null)
             : base(_assertions, system)
         {
-            if(system != null)
```
Comment (Aaronontheweb) [inline src/Akka.TestKit.NUnit/TestKit.cs:38]: Remove code bomb

File: src/Akka.TestKit.NUnit/TestKit.cs [hunk 4]
```diff
@@ -68,26 +65,14 @@ public TestKit(string config)
         public new static Config FullDebugConfig { get { return TestKitBase.FullDebugConfig; } }
 
         protected static NUnitAssertions Assertions { get { return _assertions; } }
-
-        /// <summary>
-        /// This method is called before each test run, it initializes the test including
-        /// creating and setting up the ActorSystem.
-        /// </summary>
-        [SetUp]
-        public void InitializeActorSystemOnSetUp()
```
Comment (Aaronontheweb) [inline src/Akka.TestKit.NUnit/TestKit.cs:77]: Setup no longer needed now that this can be done from `ActorSystem` constructor.

**Review by Aaronontheweb — COMMENTED** [review]
Reviewed my own changes

**Comment by mchandschuh** [comment]
@Aaronontheweb -- any estimate on this being merged/released?

File: src/Akka.TestKit.NUnit3.Tests/Bugfix44Spec.cs [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    public void Should_use_provided_ActorSystem()
+    {
+        Assertions.AssertEqual("Foo",Sys.Name);
+    }
+    
+    [Test]
+    public void Should_use_provided_ActorSystem_again()
+    {
+        // Technically this will be a second, different ActorSystem but the names should be the same
+        Assertions.AssertEqual("Foo",Sys.Name);
```
Comment (mchandschuh) [inline src/Akka.TestKit.NUnit3.Tests/Bugfix44Spec.cs:25]: Perhaps a better assertion would be storing a static to verify it's different between the test runs?

``` csharp
private static ActorSystem previousSystem;

public Bugfix44Spec() : base(ActorSystem.Create("Foo", TestKit.DefaultConfig))
{
    previousSystem ??= Sys;
}

// mark other test as [Order(1)]

[Test, Order(2)]
// nit - 'again' in test name seems misleading here, making it sound like we're reusing the same instance
public void Should_use_provided_ActorSystem_again()
{
    Asserts.AssertNotEqual(previousSystem, Sys)
}
```

Comment (Aaronontheweb) [reply src/Akka.TestKit.NUnit3.Tests/Bugfix44Spec.cs:25]: Got it - I wanted to do that but wasn't sure how with NUnit. The `Order` parameter is what I needed. Thanks. I'll give that a shot.

**Comment by Aaronontheweb** [comment]
@mchandschuh I was just waiting for your approval. Now that it's done I can push it any time.

File: src/Akka.TestKit.NUnit3.Tests/Bugfix44Spec.cs (hunk 5 above)
Comment (Aaronontheweb) [reply src/Akka.TestKit.NUnit3.Tests/Bugfix44Spec.cs:25]: Done - test passes.

**Comment by mchandschuh** [comment]
@Aaronontheweb -- I was expecting a new `Akka.TestKit.NUnit3` version but only see the new version of `Akka.TestKit.Nunit` --  do I need to switch dependencies? I never really understood the difference between the two.

**Comment by mchandschuh** [comment]
FYI -- might want to mark this as a breaking change -- anyone who was previously using `[SetUp]` must now use `[OneTimeSetUp]` which is something nunit enforces when using `InstancePerTestCase`
```
OneTimeSetUp: System.InvalidOperationException : Only static OneTimeSetUp and OneTimeTearDown are allowed for InstancePerTestCase mode.
  Exception doesn't have a stacktrace
```

**Comment by mchandschuh** [comment]
@Aaronontheweb @eaba -- thanks guys! This was successfully integrated!

As a note -- any `[OneTimeSetUp]` and `[OneTimeTearDown]` methods must now be defined as `static` -- enforced by nunit.

**Pull request merged**